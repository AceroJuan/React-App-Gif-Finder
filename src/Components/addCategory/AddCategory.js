import React from "react";
import PropTypes from "prop-types";

const AddCategory = () => {
  return (
    <>
      <h2>AddCategory</h2>
    </>
  );
};

AddCategory.propTypes = {};

export default AddCategory;
