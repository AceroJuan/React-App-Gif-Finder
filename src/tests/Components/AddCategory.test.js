import React from "react";
import "@testing-library/jest-dom";
import AddCategory from "../../Components/AddCategory/AddCategory";
import { shallow } from "enzyme";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);

  test("debe mostrar <AddCategory /> de manera correcta", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola";
    input.simulate("change", { target: { value } });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });
});
