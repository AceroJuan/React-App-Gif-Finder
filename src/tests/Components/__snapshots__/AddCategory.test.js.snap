// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrar <AddCategory /> de manera correcta 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
